diff --git a/docroot/sites/all/modules/contrib/file_entity/file_entity.file.inc b/docroot/sites/all/modules/contrib/file_entity/file_entity.file.inc
index 4949f4b69..18c1e9a6c 100644
--- a/docroot/sites/all/modules/contrib/file_entity/file_entity.file.inc
+++ b/docroot/sites/all/modules/contrib/file_entity/file_entity.file.inc
@@ -96,11 +96,11 @@ function file_entity_file_update($file) {
   }
 
   if (module_exists('image') && file_entity_file_get_mimetype_type($file) == 'image' && $file->filesize && isset($file->original)) {
-    if (!isset($file->metadata)) {
+    if (!isset($file->metadata) || gettype($file->metadata) == 'string') {
       $file->metadata = array();
     }
 
-    if (!isset($file->original->metadata)) {
+    if (!isset($file->original->metadata) || gettype($file->original->metadata) == 'string') {
       if (!is_object($file->original)) {
         $file->original = new stdClass();
       }
