diff --git a/includes/processor_highlight.inc b/includes/processor_highlight.inc
index 8accf4d..2b748f7 100644
--- a/includes/processor_highlight.inc
+++ b/includes/processor_highlight.inc
@@ -225,7 +225,10 @@ class SearchApiHighlight extends SearchApiAbstractProcessor {
     if (empty($result['entity'])) {
       return $data;
     }
-    $wrapper = $this->index->entityWrapper($result['entity'], FALSE);
+    $entities = array($result['entity']);
+    $this->index->dataAlter($entities);
+    $result['entity'] = reset($entities);
+    $wrapper = $this->index->entityWrapper($result['entity'], TRUE);
     $wrapper->language($language->language);
     $extracted = search_api_extract_fields($wrapper, $needs_extraction, array('sanitize' => TRUE));
 
@@ -312,7 +315,7 @@ class SearchApiHighlight extends SearchApiAbstractProcessor {
    */
   protected function createExcerpt($text, array $keys) {
     // Prepare text by stripping HTML tags and decoding HTML entities.
-    $text = strip_tags(str_replace(array('<', '>'), array(' <', '> '), $text));
+    $text = strip_tags(str_replace(array('<', '>'), array(' <', '> '), htmlspecialchars_decode($text)));
     $text = ' ' . decode_entities($text);
 
     // Extract fragments around keywords.
